cmd/thanos/bucket.go

```diff
@@ -406,25 +412,17 @@ func refresh(ctx context.Context, logger log.Logger, bucketUI *ui.Bucket, durati
 	})
 }
 
-func download(ctx context.Context, logger log.Logger, bkt objstore.Bucket) (blocks []metadata.Meta, err error) {
+func download(ctx context.Context, logger log.Logger, bkt objstore.Bucket, fetcher *block.MetaFetcher) (blocks []metadata.Meta, err error) {
```

Comment:
I think we can return map here and operate on map on download caller to avoid unnecessary allocations/operations. The reason is rather readability not performance.

Reply:
the caller function is marshaling the returned array, is it ok if we return map instead of array now?
Will there be any change in the caller because of this?

Reply:
In this case probably no (: ignore my comment, you are right :+1: